Add ShareableRoutablePageMixin and routed and served requests attributes (#43)
cfpb/wagtail-sharing: README.rst

This project adds these additional hooks:

``before_route_page``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Called when routing, before a page's ``route()`` method is called. This hook is passed the ``ServeView``, the ``request``, and the ``page`` that will have ``route()`` called on it. If the callable returns an ``HttpResponse``, that response will be returned immediately to the user.

Reviewer: ```suggestion
Called when routing, before a page's ``route()`` method is called. This hook is passed the ``request`` and the ``page`` that will have ``route()`` called on it. If the callable returns an ``HttpResponse``, that response will be returned immediately to the user.
```

> This hook could be useful for changing how pages that override Wagtail's built-in routing to subpages handle sharing.

Reviewer: How about adding a specific mention of routable pages here?

```suggestion
This hook allows for any necessary customization of Wagtail's built-in routing behavior, for example to support ShareableRoutablePageMixin (with link to below section of documentation).
```

> This hook allows for any necessary customization of Wagtail's built-in routing behavior, for example to support ShareableRoutablePageMixin (with link to below section of documentation).

Reviewer 2: > (with link to below section of documentation)

Is this a note to self?

Author: Yup, I think it was. That's been changed to a link in b24956c.

  def add_custom_header(page, response):
      response['Wagtail-Is-Shared'] = '1'

Mixins
------

``ShareableRoutablePageMixin``
~~~~~~~~~~~~~~~~~~~~~~~~~~~

 .. |RoutablePageMixin| replace:: ``RoutablePageMixin``
 .. _RoutablePageMixin: https://docs.wagtail.io/en/stable/reference/contrib/routablepage.html

> By default, Wagtail's |RoutablePageMixin|_ is not compatible with W-S, instead you need to use ``ShareableRoutablePageMixin`` in order to view share draft content fields on routable pages.

Reviewer: ```suggestion
By default, Wagtail's |RoutablePageMixin|_ is not compatible with Wagtail Sharing. To support routable pages, instead use ``ShareableRoutablePageMixin`` in order to view shared draft content fields.
```

> By default, Wagtail Sharing cannot share draft content on routable pages (pages that use |RoutablePageMixin|_ to define mutliple sub-URLs). To ensure that draft content fields on routable pages are sharable, Wagtail Sharing provides an subclass of |RoutablePageMixin|_, ``ShareableRoutablePageMixin``. 

Reviewer: Minor typo here: "mutliple".

More generally, the use of "by default" here might be a little confusing at first. It might be clearer to say something like "Wagtail's RoutablePageMixin is not compatible with W-S, instead you need to use ShareableRoutablePageMixin".

Author: Used the recommended language here.

> ``SharableRoutablePageMixin`` is used exactly the same way as |RoutablePageMixin|_:

Reviewer: ```suggestion
``ShareableRoutablePageMixin`` is used exactly the same way as |RoutablePageMixin|_:
```
